### ISSUE TITLE
how add a object contain a attr type is Byte.

### ISSUE THREAD

**Issue by keepcleargas** [yahoo/elide#2185]
i  got some entity like this

```java
package cn.xiaohuodui.model;

import com.yahoo.elide.annotation.Include;
import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.Set;

/**
 * @author :keepcleargas
 * @date :2021-06-29 11:13.
 */
@Include
@Entity
@Table(name = "t_course_category")
@SQLDelete(sql = "update t_course_category SET deleted = true where id=?")
//@Where(clause = "deleted=false")
public class CourseCategory {
    private int id;
    private String name;
    private String img;
    private String des;
    private Integer sort;
    private Long createAt;
    private Long updateAt;
    private Byte deleted;

    private CourseCategory parent;

    private Set<CourseCategory> children;

    @OneToMany(mappedBy = "parent")
    public Set<CourseCategory> getChildren() {
        return children;
    }

    public void setChildren(Set<CourseCategory> children) {
        this.children = children;
    }

    @ManyToOne(optional = true)
    @NotFound(action = NotFoundAction.IGNORE)
    @JoinColumn(name = "parent_id", nullable = true)
    public CourseCategory getParent() {
        return parent;
    }

    public void setParent(CourseCategory parent) {
        this.parent = parent;
    }

    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Basic
    @Column(name = "img")
    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    @Basic
    @Column(name = "des")
    public String getDes() {
        return des;
    }

    public void setDes(String des) {
        this.des = des;
    }

    @Basic
    @Column(name = "sort")
    public Integer getSort() {
        return sort;
    }

    public void setSort(Integer sort) {
        this.sort = sort;
    }

    @Basic
    @Column(name = "create_at")
    public Long getCreateAt() {
        return createAt;
    }

    public void setCreateAt(Long createAt) {
        this.createAt = createAt;
    }

    @Basic
    @Column(name = "update_at")
    public Long getUpdateAt() {
        return updateAt;
    }

    public void setUpdateAt(Long updateAt) {
        this.updateAt = updateAt;
    }

    @Basic
    @Column(name = "deleted")
    public Byte getDeleted() {
        return deleted;
    }

    public void setDeleted(Byte deleted) {
        this.deleted = deleted;
    }
}

```

---

`mutation` like this

```gql
mutation addCategory{
  courseCategory(op: UPSERT, data: {name:"hello",sort:100,deleted:{
    value:0
  }}) {
    edges {
      node {
        id
        name
        img
        sort
        deleted{
          __typename
        }
        parent {
          edges {
            node {
              id
              name
              img
            }
          }
        }
        children {
          edges {
            node {
              id
              name
              img
            }
          }
        }
      }
    }
  }
}

```
 **i got the error response**:
how can i insert `deleted` byte object to my db 
```gql
{
  "data": null,
  "errors": [
    {
      "message": "Validation error of type WrongType: argument &#39;data.deleted&#39; with value &#39;ObjectValue{objectFields=[ObjectField{name=&#39;name&#39;, value=StringValue{value=&#39;hello&#39;}}, ObjectField{name=&#39;sort&#39;, value=IntValue{value=100}}, ObjectField{name=&#39;deleted&#39;, value=ObjectValue{objectFields=[ObjectField{name=&#39;value&#39;, value=IntValue{value=0}}]}}]}&#39; contains a field not in &#39;byteInput&#39;: &#39;value&#39;",
      "locations": [
        {
          "line": 2,
          "column": 30
        }
      ]
    }
  ]
}
```

**Comment by aklish** [comment]
I would create an Elide `Serde` to register a new type for `Byte`:

https://elide.io/pages/guide/v5/09-clientapis.html#type-coercion

You can then convert Byte to a String, number, boolean or object.

**Comment by keepcleargas** [comment]
<img width="1551" alt="图片" src="https://user-images.githubusercontent.com/1215177/123888337-db6a8100-d985-11eb-804f-f47dca52a11d.png">

i just saw a `Building input object for type: java.lang.Byte`, What is this function mean

**Comment by aklish** [comment]
In the absence of a user defined Serde, Elide will attempt to convert any type it doesn't understand into an object type for GraphQL.  It will create an input and query object for each type.

**Comment by aklish** [comment]
So I ran a quick test with the spring boot example project.  I modified the `ArtifactGroup` bean as follows:

```java
@Include(name = "group")
@Table(name = "artifactgroup")
@Entity
public class ArtifactGroup {
    @Id
    private String name = "";

    private String commonName = "";

    private String description = "";

    @Transient
    @ComputedAttribute
    private Byte foo;

    @OneToMany(mappedBy = "group")
    private List<ArtifactProduct> products = new ArrayList<>();
}
```

I looked both in Graphiql and in my IDE.  In both cases, Elide doesn't identify any attributes or relationships for the Byte class.  It is an empty stub.  

For objects, Elide is looking for getter and setter methods or fields (depending on the whether the model is using properties or fields).  In this case, Byte has neither - so Elide just generates an empty stub.  

Byte is really more like a scalar.  I would recommend defining a `Serde` in this case.  
Byte might be a good addition to Elide as a standard Serde that is included for others to use. 

If you are interested in adding one, I'd be happy to review the PR.  Here is a similar PR for adding `Instant` as a standard Elide primitive type:

https://github.com/yahoo/elide/pull/1417
